presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -79,33 +79,38 @@
     private final TypeManager typeManager;
     private final int maxPartitionsPerScan;
     private final int domainCompactionThreshold;
+    private final ColumnConverterProvider columnConverterProvider;
 
     @Inject
     public HivePartitionManager(
             TypeManager typeManager,
-            HiveClientConfig hiveClientConfig)
+            HiveClientConfig hiveClientConfig,
+            ColumnConverterProvider columnConverterProvider)
     {
         this(
                 typeManager,
                 hiveClientConfig.getDateTimeZone(),
                 hiveClientConfig.isAssumeCanonicalPartitionKeys(),
                 hiveClientConfig.getMaxPartitionsPerScan(),
-                hiveClientConfig.getDomainCompactionThreshold());
+                hiveClientConfig.getDomainCompactionThreshold(),
+                columnConverterProvider);
     }
 
     public HivePartitionManager(
             TypeManager typeManager,
             DateTimeZone timeZone,
             boolean assumeCanonicalPartitionKeys,
             int maxPartitionsPerScan,
-            int domainCompactionThreshold)
+            int domainCompactionThreshold,
+            ColumnConverterProvider columnConverterProvider)
```

Comment:
You don't need to pass in `columnConverterProvider`. Ideally, we would centralize where we create and use it. For now, `MetastoreContext` and `metastore` are the ones owning the object. Given `metastore` is available in this class, you can directly get `columnConverterProvider` from it.